src/mlpack/methods/random_forest/random_forest_regressor_impl.hpp

```diff
@@ -0,0 +1,510 @@
+    template<typename> class CategoricalSplitType,
+    bool UseBootstrap
+>
+template<typename Archive>
+void RandomForestRegressor<
+    FitnessFunction,
+    DimensionSelectionType,
+    NumericSplitType,
+    CategoricalSplitType,
+    UseBootstrap
+>::serialize(Archive& ar, const uint32_t /* version */)
+{
+  size_t numTrees;
+  if (cereal::is_loading<Archive>())
+    trees.clear();
+  else
+    numTrees = trees.size();
+
+  ar(CEREAL_NVP(numTrees));
+
+  // Allocate space if needed.
+  if (cereal::is_loading<Archive>())
+    trees.resize(numTrees);
+
+  ar(CEREAL_NVP(trees));
+  ar(CEREAL_NVP(avgGain));
+}
+
+template<
+    typename FitnessFunction,
+    typename DimensionSelectionType,
+    template<typename> class NumericSplitType,
+    template<typename> class CategoricalSplitType,
+    bool UseBootstrap
+>
+template<bool UseWeights, bool UseDatasetInfo, typename MatType, typename ResponsesType>
+double RandomForestRegressor<
+    FitnessFunction,
+    DimensionSelectionType,
+    NumericSplitType,
+    CategoricalSplitType,
+    UseBootstrap
+>::Train(const MatType& dataset,
+         const data::DatasetInfo& datasetInfo,
+         const arma::Row<ResponsesType>& responses,
+         const arma::rowvec& weights,
+         const size_t numTrees,
+         const size_t minimumLeafSize,
+         const double minimumGainSplit,
+         const size_t maximumDepth,
+         DimensionSelectionType& dimensionSelector,
+         const bool warmStart)
+{
+  // Reset the forest if we are not doing a warm-start.
+  if (!warmStart)
+    trees.clear();
+  const size_t oldNumTrees = trees.size();
+  trees.resize(trees.size() + numTrees);
+
+  // Convert avgGain to total gain.
+  double totalGain = avgGain * oldNumTrees;
+
+  // Train each tree individually.
+  #pragma omp parallel for reduction( + : totalGain)
+  for (size_t i = 0; i < numTrees; ++i)
+  {
+    // NOTE: this is a hacky workaround for older versions of Armadillo that did
+    // not (by default) set a different seed for each RNG.  We simply manually
+    // set the RNG seed for each individual thread.  However, if users have
+    // specifically set the RNG seed for Armadillo, this could break their code.
+    // So we hide it behind an (undocumented except for here) ifndef so that
+    // this support can be disabled if desired...
+    #undef MLPACK_ARMA_VERSION
+    #define MLPACK_ARMA_VERSION (ARMA_VERSION_MAJOR * 100000 + \
+                                 ARMA_VERSION_MINOR *    100 + \
+                                 ARMA_VERSION_PATCH)
+    #if (MLPACK_ARMA_VERSION < 1200602) // 12.6.2
+      #ifndef MLPACK_DONT_OVERWRITE_ARMA_RNG_SEEDS
+      // Note that each thread has its own differently-seeded RNG, so this will
+      // result in a different seed for each thread's Armadillo RNG.
+      arma::arma_rng::set_seed(RandGen()());
+      #endif
+    #endif
+
+    MatType bootstrapDataset;
+    arma::Row<ResponsesType> bootstrapresponses;
```

Comment:
```suggestion
    arma::Row<ResponsesType> bootstrapResponses;
```
This is a pretty pedantic style fix, sorry for the tedium :smile: